src/engraving/tests/rhythmicgrouping_tests.cpp

```diff
@@ -103,3 +103,45 @@ TEST_F(Engraving_RhythmicGroupingTests, groupShortenNotes)
 {
     group("groupShortenNotes.mscx", "groupShortenNotes-ref.mscx");  // test for regrouping rhythms when notes should be shortened
 }
+
+TEST_F(Engraving_RhythmicGroupingTests, groupMaintainSelection)
+{
+    MasterScore* score = ScoreRW::readScore(RHYTHMICGRP_DATA_DIR + String::fromUtf8("groupSubbeats.mscx"));
+    ASSERT_TRUE(score);
+
+    score->cmdSelectAll();
+
+    const Selection& selection = score->selection();
```

Comment:
Thanks for those tweaks!

One last thing - we normally include comments for tests to explain what's going on (I know it's fairly self explanatory here, but still worth doing).

Something like this (for both tests):

```
// [GIVEN] A score with a selection
...
// [WHEN] Regrouping rhythms
...
// [EXPECT] Selection is restored
```

Reply:
No worries! I've added some comments to the tests I added. Looking at the other engraving tests, I also realized I never deleted the score, which I've also fixed.

Reply:
Ah yes, good catch - approved!